modules/gmake/gmake.lua

```diff
@@ -0,0 +1,338 @@
+---
+-- GNU Makefile exporter.
+---
+
+local dom = require('dom')
+local path = require('path')
+local premake = require('premake')
+local State = require('state')
+
+local gmake = {}
+
+gmake.wks = doFile('./src/wks.lua', gmake)
+gmake.proj = doFile('./src/proj.lua', gmake)
+
+local _ARCHITECTURES = {
+	x86 = 'x86',
+	x86_64 = 'x86_64',
+	arm = 'ARM',
+	arm64 = 'ARM64'
+}
+
+
+---
+-- Table of all toolsets.
+---
+gmake.toolsets = {
```

Comment:
Shouldn't those toolset specific stuff be moved in the appropriate toolset directly?

Reply:
Very much agreed, just waiting for the toolset API then I'll do it.

Reply:
@Jarod42 We weren't really as clear as we could have been on the intention of this PR. We're just trying to bootstrap a 6.x makefile exporter up to the same level as the current 6.x VS exporter: a minimal implementation able to generate a barebones but working makefile for Premake itself. I'm also working an Xcode version of the same thing. Some things—like toolsets and everything related to them—will be placeholders until we have this baseline work in place.